src/alchemlyb/estimators/ti_gaussian_quadrature_.py

```diff
@@ -0,0 +1,228 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+from .. import concat
+from .base import _EstimatorMixOut
+
+
+class TI_GQ(BaseEstimator, _EstimatorMixOut):
+    """Thermodynamic integration (TI) with gaussian quadrature estimation.
+       When the simulations are performed at certain gaussian quadrature
+       points (lambdas), the free energy can be estimated using gaussian 
+       quadrature as an alternative to the trapezoidal rule.
+
+    Parameters
+    ----------
+
+    verbose : bool, optional
+        Set to True if verbose debug output is desired.
+
+    Attributes
+    ----------
+
+    delta_f_ : DataFrame
+        The estimated cumulative free energy from one state to another.
+
+    d_delta_f_ : DataFrame
+        The estimated statistical uncertainty (one standard deviation) in
+        dimensionless cumulative free energies.
+
+    states_ : list
+        Lambda states for which free energy estimation were obtained.
+
+    dhdl : DataFrame
+        The estimated dhdl of each state.
+
+
+    .. versionadded:: 2.0.0
+       `delta_f_`, `d_delta_f_`, `states_` are view of the original object.
+
+    """
+
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+    def fit(self, dHdl):
+        """
+        Compute cumulative free energy from one state to another by integrating
+        dHdl across lambda values.
+
+        Parameters
+        ----------
+        dHdl : DataFrame
+            dHdl[n,k] is the potential energy gradient with respect to lambda
+            for each configuration n and lambda k.
+
+        """
+
+        # sort by state so that rows from same state are in contiguous blocks,
+        # and adjacent states are next to each other
+        dHdl = dHdl.sort_index(level=dHdl.index.names[1:])
+
+        # obtain the mean and variance of the mean for each state
+        # variance calculation assumes no correlation between points
+        # used to calculate mean
+        self.means = dHdl.groupby(level=dHdl.index.names[1:]).mean()
+        self.variances = np.square(dHdl.groupby(level=dHdl.index.names[1:]).sem())
+        # suggested lambda values and corresponding weights 
+        special_points = self.get_quadrature_points()
+        weights = []
+        # check if the lambdas in the simulations match the suggested values
+        lambda_list, means_list, variances_list = self.separate_dhdl()
+        for lambdas in lambda_list:
+            num_lambdas = len(lambdas)
+            if (num_lambdas not in special_points) or (not np.allclose(lambdas, special_points[num_lambdas][0], rtol=0.1)):
+                raise ValueError('The lambda values for gaussian quadrature are not supported, please use trapezoidal rule instead.')
+            weights.extend(special_points[num_lambdas][1])
```

Comment:
This `[1]` is a bit confusing. Might just make `special_points` as a class attribute in the format of
`special_points = {1: {'lambdas': [], 'weights': []}, 2: {'lambdas': [], 'weights': []},}`